add a read me doc for wechat pay module (#4052)
stripe/stripe-android: wechatpay/README.md

# Stripe Android SDK WeChatPay module

> Module to provide support for `PaymentIntent` with `wechat_pay` as its payment method.

Reviewer: ```suggestion
This module provides support for `PaymentIntent` with `wechat_pay` as its payment method.
```

 
# Overview

> * This module will provide a `PaymentAuthenticator` implementation to authenticate a [PaymentIntent](https://stripe.com/docs/api/payment_intents) with [next_action.wechat_pay_redirect_to_android_app](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-next_action-wechat_pay_redirect_to_android_app) by invoking the WeChat Pay SDK and pass the required parameters from that hash.

Reviewer: ```suggestion
* This module will provide a `PaymentAuthenticator` implementation to authenticate a [PaymentIntent](https://stripe.com/docs/api/payment_intents) with [next_action.wechat_pay_redirect_to_android_app](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-next_action-wechat_pay_redirect_to_android_app) by invoking the WeChat Pay SDK and passing the required parameters from that hash.
```
```suggestion
* This module will provides a `PaymentAuthenticator` implementation for authenticating .a [PaymentIntent](https://stripe.com/docs/api/payment_intents).  To do this it invokes the WeChat Pay SDK to retrieve a hash, and sets this value in the [PaymentIntent.next_action.wechat_pay_redirect_to_android_app](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-next_action-wechat_pay_redirect_to_android_app).
```

Reviewer: I tried splitting the sentence up to make it a bit clear.  I tried to make it so it first stating that it provides and Authenticator and then, in a second sentence, stating what it does.   

I am having trouble editing the suggestion now.  There is a '.' after the word authenticating that should be removed.

* No extra code is needed when using this module, however you will need to declare dependency on the actual WeChat Pay SDK, as this module uses reflection to invoke it.

# Usage
* In app/build.gradle, add these dependencies:
    ```gradle
    dependencies {
      // Stripe dependencies, make sure they have the same $stripe_sdk_version
      implementation 'com.stripe:stripe-android:$stripe_sdk_version' // Main Stripe SDK
      implementation 'com.stripe:stripe-wechatpay:$stripe_sdk_version' // WeChat Pay module
      // WeChat Pay SDK, make sure 6.7.0 is used
      implementation 'com.tencent.mm.opensdk:wechat-sdk-android-without-mta:6.7.0'
    }
    ```

> * Then in your app, just confirm and capture the result of the `PaymentIntent` with WeChat Pay like others(see an complete example [here](https://stripe.com/docs/payments/integration-builder)), WeChat app on the phone will be opened to confirm the payment.

Reviewer: ```suggestion
* Then in your app, just confirm and capture the result of the `PaymentIntent` with WeChat Pay like others (see a complete example [here](https://stripe.com/docs/payments/integration-builder)), the WeChat app on the phone will be opened to confirm the payment.
```

    ```kotlin
    class CheckoutActivity : AppCompatActivity() {

>         private fun startCheckout() {

Reviewer: In the [past](https://stripe.com/docs/payments/accept-a-payment?platform=android) we have put this in onCreate

Author: ditto, removed the code unrelated to wechat

>             payButton.setOnClickListener {

Reviewer: Being already familiar with the SDk I know the type of cardInputWidget, but someone new to the SDK might not be familiar. 

It might be helpful to include a simple xml, or a comment describing the variables.  In (this)[https://stripe.com/docs/payments/accept-a-payment?platform=android] example we show them as member variables in the class and then have a comment that instructs them to instantiate it.

Author: trying to just focus on wechat specific context here, removed the unrelated part

                // create paymentMethod with WeChat Pay type

>                 val paymentMethodCreateParams = createWeChatPayParams()

Reviewer: I am curious why you chose not to add the createWeChatPayParams() function in this code snippet?

                // confirm the intent with the param
                cardInputWidget.paymentMethodCreateParams?.let { params ->
                    val confirmParams = ConfirmPaymentIntentParams
                        .createWithPaymentMethodCreateParams(params, paymentIntentClientSecret)
                    stripe.confirmPayment(this, confirmParams)
                }
            }
        }

        override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
            super.onActivityResult(requestCode, resultCode, data)
            if (stripe.isPaymentResult(requestCode, data)) {
                lifecycleScope.launch {
                    runCatching {
                        stripe.getPaymentIntentResult(requestCode, data!!).intent
                    }.fold( /* retrieve success/failure result */ )
                }
            }
        }
    }
    ```

# Notice

> 1. Make sure your Android app is registered in [WeChat Open Platform](https://open.weixin.qq.com/) and its package name(应用包名) from `AndroidManifest.xml` and signature(应用签名) correctly uploaded.

Reviewer: ```suggestion
1. Make sure your Android app is registered in [WeChat Open Platform](https://open.weixin.qq.com/) and its package name (应用包名) from `AndroidManifest.xml` and signature (应用签名) correctly uploaded.
```

    > Note: Use the __GenSignature__ tool provided by WeChat [here](https://pay.weixin.qq.com/wiki/doc/api/app/app.php?chapter=8_5) to generate a signature, note the signature will differ from debug build to release build.
2. Make sure the WeChat app installed on your test phone has a WeChat account with WeChat Pay enabled.
3. Make sure to create a Stripe `PaymentIntent` with a Stripe _live_ key - the `PaymentIntent` created by a _test_ key will have dummy WeChat Pay parameters that won't be recognized by WeChat.

> 4. Apart from adding the dependencies, you __don't__ need to write any additional code with WeChat Pay SDK(E.g the `WXPayEntryActivity` or any callbacks from the WeChat Pay [doc](https://pay.weixin.qq.com/wechatpay_guide/help_docs.shtml)), the module will configure those for you.

Reviewer: ```suggestion
4. Apart from adding the dependencies, you __don't__ need to write any additional code with WeChat Pay SDK (e.g the `WXPayEntryActivity` or any callbacks from the WeChat Pay [doc](https://pay.weixin.qq.com/wechatpay_guide/help_docs.shtml)), the module will configure those for you.
```
